aten/src/ATen/native/cpu/UpSampleKernel.cpp

```diff
@@ -1134,31 +1332,35 @@ void upsample_generic_Nd_kernel_impl(
   }
 }
 
-template <typename scalar_t>
-void cpu_upsample_generic_aa(at::TensorIterator& iter) {
+template <typename scalar_t, bool is_horizontal>
+void cpu_upsample_generic_aa(at::TensorIterator& iter, unsigned int weights_precision) {
 
   auto loop = [&](char** data, const int64_t* strides, int64_t n) {
-    if ((strides[0] == sizeof(scalar_t)) && (strides[1] == sizeof(scalar_t)) &&
-        is_zero_stride<3 + 2>(&strides[2])) {
-      basic_loop_aa_single_dim_zero_strides<scalar_t, int64_t>(
-          data, strides, n);
+    if (is_horizontal) {
```

Comment:
I think I have to return back if conditions on the stride for auto vectorization

Reply:
BTW, I noticed that with compiler optimization level `-O2`, auto-vectorization may not work as intended with gcc. I'm guessing you've already verified that auto-vectorization works well here? Thanks!

Reply:
Thanks for the note @sanchitintel. Yes I agree that with `O2` auto-vec can work less good and definitely worth checking.

Reply:
Hi @NicolasHug, the release binaries [seem to be built with the `-O2` compilation flag](https://github.com/pytorch/pytorch/blob/master/CMakeLists.txt#L813). I don't think [PyTorch Builder](https://github.com/pytorch/builder) changes compilation flags before building.

Reply:
Hi @sanchitintel I would say it is a bit unclear as https://github.com/pytorch/pytorch/blob/7963dbf3db60123ae18c203265ea2e89457cd2fe/cmake/Codegen.cmake#L60 defines flags for ATen build and also inspecting in the local build (without explicitly overriding `CXXFLAGS`) the following file `build/compile_commands.json`, we can see the flags for individual files:
```
{
  "directory": "/pytorch/build",
  "command": "/usr/bin/c++ -DAT_PER_OPERATOR_HEADERS -DCPUINFO_SUPPORTED_PLATFORM=1 -DFMT_HEADER_ONLY=1 -DHAVE_MALLOC_USABLE_SIZE=1 -DHAVE_MMAP=1 -DHAVE_SHM_OPEN=1 -DHAVE_SHM_UNLINK=1 -DMINIZ_DISABLE_ZIP_READER_CRC32_CHECKS -DONNXIFI_ENABLE_EXT=1 -DONNX_ML=1 -DONNX_NAMESPACE=onnx_torch -DUSE_EXTERNAL_MZCRC -D_FILE_OFFSET_BITS=64 -Dtorch_cpu_EXPORTS -Iaten/src -I../aten/src -I. -I../ -I../third_party/onnx -Ithird_party/onnx -I../third_party/foxi -Ithird_party/foxi -I../torch/csrc/api -I../torch/csrc/api/include -I../caffe2/aten/src/TH -Icaffe2/aten/src/TH -Icaffe2/aten/src -Icaffe2/../aten/src -I../torch/csrc -I../third_party/miniz-2.1.0 -I../third_party/kineto/libkineto/include -I../third_party/kineto/libkineto/src -I../aten/../third_party/catch/single_include -I../aten/src/ATen/.. -I../c10/.. -I../third_party/pthreadpool/include -I../third_party/cpuinfo/include -I../aten/src/ATen/native/quantized/cpu/qnnpack/include -I../aten/src/ATen/native/quantized/cpu/qnnpack/src -I../third_party/cpuinfo/deps/clog/include -I../third_party/ittapi/src/ittnotify -I../third_party/FP16/include -I../third_party/fmt/include -I../third_party/flatbuffers/include -isystem ../third_party/protobuf/src -isystem ../third_party/ittapi/include -isystem ../cmake/../third_party/eigen -isystem include  -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DLIBKINETO_NOROCTRACER -DUSE_PYTORCH_QNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow -DHAVE_AVX2_CPU_DEFINITION -O3 -DNDEBUG -DNDEBUG -fPIC -DTH_HAVE_THREAD -Wall -Wextra -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-type-limits -Wno-array-bounds -Wno-sign-compare -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-missing-braces -Wno-maybe-uninitialized -fvisibility=hidden -O2 -fopenmp -DCAFFE2_BUILD_MAIN_LIB -pthread -std=gnu++17 -O3  -mavx2 -mfma -mno-avx256-split-unaligned-load -mno-avx256-split-unaligned-store -DCPU_CAPABILITY=AVX2 -DCPU_CAPABILITY_AVX2 -o caffe2/CMakeFiles/torch_cpu.dir/__/aten/src/ATen/native/cpu/UpSampleKernel.cpp.AVX2.cpp.o -c /pytorch/build/aten/src/ATen/native/cpu/UpSampleKernel.cpp.AVX2.cpp",
  "file": "/pytorch/build/aten/src/ATen/native/cpu/UpSampleKernel.cpp.AVX2.cpp"
},
```
and we see both `-O2` and `-O3` flags. So, I'd say O3 is being used.

Reply:
Thanks for clarifying, @vfdev-5! :)